packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,71 @@
+const fs = require("fs");
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+
+class DotenvWebpackPlugin {
+  constructor(config = {}) {
+    const currentDirectory = process.cwd();
+
+    this.config = Object.assign(
+      {},
+      {
+        envFiles: [
+          `${currentDirectory}/.env.example`, // loaded in all cases
+          `${currentDirectory}/.env`, // loaded in all cases
+          `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+          `${currentDirectory}/.env.[mode]`, // only loaded in specified mode
+          `${currentDirectory}/.env.[mode].local`, // only loaded in specified mode, ignored by git
+        ],
+        prefixes: ["process.env.", "import.meta.env."],
+      },
+      config,
+    );
+  }
+
+  apply(compiler) {
+    const mode = compiler.options.mode || "production";
+
+    // .local file variables will get precedence
+    let environmentFiles = Array.isArray(this.config.envFiles)
+      ? this.config.envFiles
+      : [this.config.envFiles];
+
+    environmentFiles = environmentFiles.map((environmentFile) =>
+      environmentFile.replace(/\[mode\]/g, mode),
+    );
+
+    let envVariables = {};
+    // parse environment vars from .env files
+    environmentFiles.forEach((environmentFile) => {
+      if (fs.existsSync(environmentFile)) {
+        try {
+          const environmentFileContents = fs.readFileSync(environmentFile);
+          const parsedEnvVariables = dotenv.parse(environmentFileContents);
+          for (const [key, value] of Object.entries(parsedEnvVariables)) {
+            // only add variables starting with WEBPACK_
+            if (key.startsWith("WEBPACK_")) {
```

Comment:
Why only `WEBPACK_`, we need to load all ENV variables

Reply:
if any variable with sensitive information gets added to any env file, it will be shipped with the client bundle. having `WEBPACK_` at the start of the variable makes sure that consumers are aware that this variable will be added to the build. 

I've seen this pattern implemented in other bundlers and Create React App, and hence, ported it here. Can remove if not required

Reply:
It means it should be configurable, some application can be written with env without `WEBPACK_` prefix and we can't force them to use `WEBPACK_` prefix